libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -504,6 +505,106 @@ TEST_F(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+// A mock class that simulate a client of the includer.
+class TestIncluder : public shaderc::CompileOptions::Includer {
+ public:
+  // Create a fake file object, with specified path and content.
+  struct FakeFile {
+    std::string path;
+    std::string content;
+  };
+  // Use hashmap as a fake file system to store fake files to be included.
+  using FakeFS = std::unordered_map<std::string, FakeFile>;
+
+  TestIncluder(FakeFS& fake_fs) : fake_fs_(fake_fs){};
+
+  // Get path and content from the fake file system.
+  shaderc_includer_response* GetIncluderResponse(
```

Comment:
You can't just reuse the same response object.  The client might request many independent or stacked includes before releasing any of them.